### PR TITLE
Block  selection

### PR DESCRIPTION
**Pull request by hexnickk**

### CODE REVIEW

File: src/stores/selected/selected.store.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+const initialState: SelectedState = [];
+export const $selectedStore = selectedDomain.createStore<SelectedState>(
+    initialState,
+    {
+        name: 'Selected store',
+    }
+);
+
+export const isSelected = (node: Element) => (state: SelectedState) =>
+    state.indexOf(node) !== -1;
```
Comment (rayman1104) [inline src/stores/selected/selected.store.ts]:
```suggestion
    state.includes(node);
```

File: src/stores/selection/selection.init.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import { resetFocus } from '~src/stores/focused';
+import { $selectedStore, setSelected } from '~src/stores/selected';
+
+const isEqSelectedArrays = (array1: readonly any[], array2: readonly any[]) =>
+    array1.length !== array2.length
+        ? false
+        : array1.reduce(
+              (acc, _, index) => array1[index] === array2[index] && acc,
+              true
+          );
```
Comment (rayman1104) [inline src/stores/selection/selection.init.ts]:
```suggestion
const arraysEqual<T> = (arr1: readonly T[], arr2: readonly T[]) =>
    arr1.length == arr2.length && 
    arr1.every((v1, i) => v1 === arr2[i]);
```

File: src/stores/focused/focused.store.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        name: 'Focused store',
+    }
+);
+
+export const isFocused = (block: Block) =>
+    $focusedStore.map((state) => state?.id === block?.id);
+
+const min = (a, b) => (a < b ? a : b);
+const max = (a, b) => (a > b ? a : b);
```
Comment (rayman1104) [inline src/stores/focused/focused.store.ts:23]: why not `Math.min`, `Math.max`?

**Review by rayman1104 — APPROVED** [review]
Looks good 👍 
Just nit comments.

File: src/stores/focused/focused.store.ts (hunk 3 above)
Comment (hexnickk) [reply src/stores/focused/focused.store.ts:23]: Yeah, reasonable, will update 👌

**Pull request merged**